design_documents/test_design.md

```diff
@@ -18,13 +17,33 @@ Propose a solution of test design that solves most of the aforementioned issues.
 
 Tests should be contracts (just like in the early days of Protostar).
 Basically, the code in test should behave as if it was a function called in an invoke transaction.
-This clears up the confusion concerning gas estimation — we just calculate the cost of the whole test 
-as it is all executed in the Starknet context. Moreover, starknet functions and system calls now can be made available
-in tests directly. Last but not least, the confusion concerning what actually happens when using dispatchers will be cleared:
-we just call another contract as if we were in invoke transaction.
+
+Solutions to aforementioned problems:
+1. Since everything happens in the Starknet context now, we can just calculate the cost of the whole test. 
+2. Starknet functions and system calls can be made available in tests directly since tests, being contracts, have their own states.
+3. See next [section](#testing-internal-functions).
+4. The confusion concerning what actually happens when using dispatchers is cleared: we just call another contract as if we were in an invoke transaction.
+
+## Testing internal functions
+
+When implementing contracts, a function called `contract_state_for_testing` is created just before compilation to Sierra/CASM, just like with dispatchers.
+It can be used to obtain the state (storage) of the current (to be tested) contract. Since each test is a contract, now internal functions can be tested
+using the storage of the test.
+
+```rust
+use pkg::HelloStarknet::debtContractMemberStateTrait;
+use pkg::HelloStarknet::balanceContractMemberStateTrait;
+
+#[test]
+fn test_essa() {
```

Comment:
Nit but let's name it something more serious.

Reply:
Leftover, thanks